### DISCUSSION TITLE
What is the difference between session.db.update(myObject) and MyClass.update(session, myObject)

### DISCUSSION THREAD

**Discussion by mozexty** [serverpod/serverpod#434] [Q&A] [answered]
In documentation and sample codes, I see two ways of doing crud operations :
`session.db.update(myObject)` and `MyClass.update(session, myObject)`.
What is the difference ?

**Comment by yahu1031** [comment, marked as answer]
@mozexty Actually, both are the same. When you go inside the generated code of `MyClass.update` method, You will be finding the implementation of `session.db.update` method.

> Which you can use with a session anywhere.
```dart
var authModel = AuthModel(status: true, statusCode: 10, message: 'message');
// using session
await session.db.update(authModel);
// using model
await AuthModel.update(session, authModel);
```

> Generated code
```dart
static Future<bool> update(
  _i1.Session session,
  AuthModel row, {
  _i1.Transaction? transaction,
}) async {
  return session.db.update(
    row,
    transaction: transaction,
  );
}
```

Finally, both are the same, you can use anything you want.

**Reply by mozexty** [reply]
@yahu1031 Thank you
